akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -99,6 +99,15 @@ object Sink {
   def foreach[T](f: function.Procedure[T]): Sink[T, CompletionStage[Done]] =
     new Sink(scaladsl.Sink.foreach(f.apply).toCompletionStage())
 
+  /**
+   * A `Sink` that will invoke the given procedure asynchronously for each received element. The sink is materialized
+   * into a [[java.util.concurrent.CompletionStage]] which will be completed with `Success` when reaching the
+   * normal end of the stream, or completed with `Failure` if there is a failure signaled in
+   * the stream.
+   */
+  def foreachAsync[T](parallelism: Int)(f: function.Function[T, CompletionStage[Void]]): Sink[T, CompletionStage[Done]] =
```

Comment:
We haven't used `CompletionStage[Void]` anywhere else. Wouldn't it be more consistent to use `CompletionStage[Done]`?

Reply:
Actually I don't think it applies in this case, I'll tell why.
If we use CompletionStage[Done]:

```java
final CompletableFuture<Done> doneCompletableFuture = CompletableFuture
  .completedFuture("a")
  .thenAccept(v -> {
      //accept the data
  })
  .thenApply(v -> Done.done());
```
Don't know about you, but, this last line just to map to Done annoys me a lot. =P haha

If we use Void:

```java
final CompletableFuture<Void> voidCompletableFuture = CompletableFuture
  .completedFuture("a")
  .thenAccept(v -> {
    //accept the data
  });
```
No need to do unnecessary maps. =)

Also, I'm using Future[Unit] in the scaladsl, so, semantically I guess it'd be better to use Void in the javadsl.

But, if you have another point I'd be glad to hear.

Reply:
Wasn't `Done` also introduced because `Void` doesn't have an instance (you can only pass `null`), and `null` isn't allowed inside the reactive streams spec (nor as materialized values)?

So introducing `Void` in the API would limit the ability for that part to become a stream or materialized value later on. Even though a `Source[Done]` would be rather silly :-)

Reply:
@patriknw in this case you wouldn't use `CompletableFuture.supplyAsync`, rather you should use `CompletableFuture.runAsync`:

```java
final Sink<Integer, CompletionStage<Done>> sinkCompletion = 
    Sink.foreachAsync(10, param -> CompletableFuture.runAsync(System.out::println));
```

Reply:
I’m convinced. We should see this as foreach.